src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -0,0 +1,1126 @@
+{
+  int count = 0 ;
+  for (unsigned int i = 0; i < in.size(); ++i)
+  {
+    if(count == 10 )
+    {
+      std::cout << std::endl; 
+      count = 0;
+    }
+    std::cout << std::setw(5) << std::left << in[i];
+    count++;
+  }
+  std::cout << std::endl;
+}
+void print(std::vector<bool> const &in)
+{
+  int count = 0 ;
+  for (unsigned int i = 0; i < in.size(); ++i)
+  {
+    if(count == 10 )
+    {
+      std::cout << std::endl; 
+      count = 0;
+    }
+    std::cout << std::setw(5) << std::left << in[i];
+    count++;
+  }
+  std::cout << std::endl    << std::endl;
+}
+
+void print( ArrayOfSets<localIndex> const & in )
+{
+  for( unsigned int i = 0; i < in.size(); ++i) {
+    for(unsigned int j = 0; j < in.sizeOfSet(i);++j)
+    {
+      std::cout << std::setw(5) << std::left << in(i,j);
+    }
+    std::cout << std::endl;
+  }
+}
+
+void print(array2d<localIndex> const &in)
+{
+  for (unsigned int i = 0; i < in.size(0); ++i)
+  {
+      std::cout << std::setw(5) << std::left << in(i,0);
+      std::cout << std::setw(5) << std::left << in(i,1);
+  }
+  std::cout << std::endl
+            << std::endl;
+}
+void print(std::vector<EdgeInfo> const &in)
+{
+  for (unsigned int  i = 0; i < in.size(); ++i)
+  {
+    std::cout << std::setw(5) << i
+              << std::setw(5) << std::left << in[i].first 
+              << std::setw(5) << std::left << in[i].second 
+              << std::endl;
+  }
+  std::cout << std::endl
+            << std::endl;
+}
+
+/***************************************************************************************/
+/***************************************************************************************/
+
+
+/**
+ * @class MeshConnectivityBuilder
+ * @brief The MeshConnectivityBuilder to build the connectivity maps 
+ * 
+ * Initially designed for hexahedral meshes (unstructured)
+ * TODO How do we reuse this for other type of cells 
+ * Most of the code will be the same 
+ * Is templating an option? Or derivation? 
+ * 
+ * TODO and here is ONE problem: all mapping is toward Elements are not safe 
+ * since the elements may not be in the same CellBlock
+ * Check what was done - Where is this used and what for?
+ *
+ * TODO Why are storage strategies different for the mappings ?
+ * TODO Why multidimensional arrays? Isn't is more expensive? 
+ * TODO Implement specialization for regular hex mesh
+ * 
+ * TODO The storage of Faces and Edges is dependant on the Cell Types  
+ * How do we manage the CellBlocks with different types of cells?
+ * Strategy 1: allocate the space for hexahedra and keep a lot of invalid stuff 
+ *             in these vectors - not the worst idea since these meshes should be hex-dominant
+ *             do we need to store the cell type?
+ * TODO For full tetrahedral meshes we need a dedicated implementation
+ * 
+ * Options : Store if the face is a triangle or quad ? 
+ *  Store a  NO_FACE  value if  same storage space for hybrid meshes
+ *  
+ */
+class MeshConnectivityBuilder
```

Comment:
The blinding curtain is already there and is called the `CellBlockManagerABC`: nothing is exposed except it. Maybe we don't we need another one?
But let's see how things evolve and we'll be able to figure out the best strategy.

Reply:
Isn't everything that is in a header file exposed?

Reply:
> Isn't everything that is in a header file exposed?

Potentially. But today, outside of the _"meshing zone"_, only `CellBlock*ABC` are exposed, since it should not be allowed to include any other header. We'll see at the end if it's worth the case.